lint/rule.go

```diff
@@ -14,10 +14,295 @@
 
 package lint
 
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/jhump/protoreflect/desc"
+)
+
 // Rule defines a lint rule that checks Google Protobuf APIs.
 type Rule interface {
```

Comment:
Please consider that we will add rules for OpenAPIs -- it is better to change the name to `ProtoRule`.

Reply:
I doubt that ever happens in real life, but done regardless.